### DISCUSSION TITLE
🏆 시즌 2  스터디 후기

### DISCUSSION THREAD

**Discussion by Ryan-Dia** [Next-by-Next/Javascript-Deep-Dive-Study#1278] [후기]
# 🎯 시즌 2 스터디 

### 📅 운영 기간 23.10.28 ~  24.01.26  ( 3개월 )

<br>

## 🏆 전체 개근상 (part 1 ~ part 3 통합)

🏆 나세현
🏆 이나린

## 👑 전체 완주자 (part 1 ~ part 3 통합)

👑 강철원
👑 한수지

<br>

# 🏅 시즌 2 스터디 후기

- 이름 :
- 참여한 part : 

### 1️⃣ 스터디 하면서 도움이 되었던 부분은?

<br>

### 2️⃣ 스터디 하면서 힘들었던 부분은?

<br>

### 3️⃣ 스터디하고나서 느낀점 및 후기

<br>

### 4️⃣ 하고 싶은 말  & 다음에는 이런 게 추가/보완되었으면 좋겠어요

**Comment by e6d1fe** [comment]
- 이름: 나세현
- 참여한 part: 1, 2, 3

#### 1️⃣ 스터디 하면서 도움이 되었던 부분은?

어떻게 읽기 시작해야 할지 엄두조차 안 나던 두꺼운 책을 드디어 읽을 수 있었습니다. 😀 읽으면서 자바스크립트라는 언어가 어떻게 동작하는지에 대해 조금이나마 더 잘 이해할 수 있게 되었습니다.

#### 2️⃣ 스터디 하면서 힘들었던 부분은?

연말에 여행이 두 번 있었는데, 개근을 놓치고 싶지 않아 노트북을 챙겨 가서 공부했습니다. 그래도 덕분에 목표였던 스터디 완주를 달성하게 되었네요! 이외에는 힘들었던 부분은 딱히 없었습니다.

#### 3️⃣ 스터디 하고 나서 느낀점 및 후기

자바스크립트를 좋아하는 편이라서 이 언어를 더 제대로 이해해보고 싶다! 라는 마음이 컸는데, 스터디 덕분에 그 목표에 한발짝 더 다가선 것 같아서 좋습니다.

#### 4️⃣ 하고 싶은 말 & 다음에는 이런 게 추가/보완되었으면 좋겠어요

혼자 읽으려고 했다면 중간에 흐지부지되거나 포기했을 게 뻔한데, part 1부터 같이 달려주신 분들이 계셔서 완주할 수 있었던 것 같습니다. 특히나 스터디 레포 관리해 주시고, 매번 디코 채널로 공지 보내주시고 심지어 중간에 이벤트까지 열어 주시면서 열심히 스터디장 역할을 해 주신 철원님께 감사하다는 말씀 전하고 싶어요! 모두 고생하셨습니다 👍🏻
이번에 중요한 부분을 먼저 짚고 넘어가느라 조금씩 빠진 부분이 있는 걸로 알고 있는데, 다음에는 스터디가 조금 길어지더라도 모든 범위를 커버해 보는 건 어떨까 싶습니다. (저는 혼자 읽고 마무리할 예정입니다 😀)

**Comment by nincoding** [comment]
- 이름: 한수지
- 참여한 part: 1, 2, 3

1️⃣ **스터디 하면서 도움이 되었던 부분은?**

개인적으로, 자바스크립트 관련 서적 중 Deep Dive를 가장 좋아해서 스터디에 참여하게 되었습니다.

혼자 학습할 때에는 책 자체가 비교적 두꺼운 분량이기에,
모든 분량 중에서도 학습이 필요한 중요 챕터를 구분하고, 
1회 완독을 위한 적절한 학습 시간과 기간을 상세히 계획하기가 쉽지 않았습니다.

하지만, 스터디에서 part 1, 2, 3로 스터디 기간을 1달 단위로 나눠,
중요한 챕터 위주로 구성된 범위를 제공받고, 일일 기준 1시간 정도의 학습 시간으로 상세히 분리된 TIL목표를 공유 받았기에 
3개월동안 부담감없이 효율적으로 완독을 할 수 있었습니다.

PR 제출 시간도 오전6시 기준이였기 때문에, 다른 학습과 병행하거나 다른 업무를 수행하고 공부하기에 큰 무리가 없었던 것 같습니다.

추가적으로 개인 학습을 마치고, 다른 스터디원분들의 PR 문서를 보면서
같은 범위를 학습하더라도 미처 생각하지 못했던 부분들 혹은 중요한 핵심 개념을 다시 한번 더 체크할 수 있었던 것 같습니다.

part 1, 2에서는 자바스크립트의 기본적인 개념들을 다시 한번 더 복습하고,
part 3에서는 자바스크립트의 동작 원리, 브라우저, 핵심 개념들에 대해 깊이 있게 이해할 수 있었기에 

이번 스터디 참여가 저에게 많은 도움이 되었던 것 같습니다.

2️⃣ **스터디 하면서 힘들었던 부분은?**

특정 프로퍼티의 개념들에 대해 반복적으로 나열된 정의 식으로만 구성된 챕터를 문서로 정리할 때 약간의 현타가 왔던 것을 제외하고는
딱히 없었던 것 같습니다.


3️⃣ **스터디 하고 나서 느낀점 및 후기**

Deep Dive는 분량 자체가 많은 만큼, ES6 이후와 이전 개념을 학습하기에 좋은 것 같습니다.
특히 part 3에 포함된 범위들이 상당히 재미있었습니다. 👍

이번 시즌2 스터디가 part 3를 기점으로 마지막이라는 점이 한편으로는 아쉬움이 있었지만,
자바스크립트 언어 자체에 조금 더 흥미가 생기게 된 계기가 된 것 같아 뿌듯함이 듭니다.

개인적으로 Deep Dive에서 포함되어 있지 않은 내용들, 
혹은 ES2015+ 이후 최신 자바스크립트의 업데이트 내용들에도 궁금증이 생겨서

추가적으로 공식 문서 등을 참고하면서,  주기적으로  Deep Dive완독을 한다면 
좋은 시너지가 될 것 같다는 생각이 들었습니다.

4️⃣ **하고 싶은 말 & 다음에는 이런 게 추가/보완되었으면 좋겠어요**

먼저, 스터디 운영을 해주시고 공지, 일정 및 레포 관리 등 끝까지 책임져주신 스터디장님께 감사드립니다. 👍
그리고 함께 스터디를 끝마치신 모든 스터디원분들 모두 고생 많으셨습니다 👏👏👏

만약 다음 시즌3 Deep Dive 스터디가 Open된다면 참여하고 싶을 정도로, 많이 자극을 받고 도움이 된 것 같습니다  :)

**Comment by narinn-star** [comment]
- 이름 : 이나린
- 참여한 part : 1, 2, 3

1️⃣ 스터디 하면서 도움이 되었던 부분은?

자바스크립트 공부를 깊게 해보고자 구매 했던 서적, 그리고 너무 딱 맞아들었던 스터디 시작 시기 덕분에 꾸준함을 가지고 공부할 수 있었습니다.
어쩌면 헛되이 보낼 수 있었던 시간을 스터디에 투자하며 의미있게 만들 수 있었습니다.

2️⃣ 스터디 하면서 힘들었던 부분은?

크게 이 스터디가 너무 힘들다 했던 부분은 없었습니다. 
그저 저의 나태함이 조금 힘들었어요..

3️⃣ 스터디하고나서 느낀점 및 후기

단순히 자바스크립트를 더 자세히, 더 많이 알고싶어 시작한 스터디로 자바스크립트를 더 깊이 알고싶게 되기도 했고, 
기본 지식들도 많이 담아가는 것 같습니다 :)

4️⃣ 하고 싶은 말 & 다음에는 이런 게 추가/보완되었으면 좋겠어요 

이 두꺼운 책을 혼자 보려했다면 아마 아직 반도 못 봤을 것 같습니다. 
어떤 날은 처음부터 꼼꼼히 읽었다면, 또 어떤 날은 이해가 가지 않아 모두 정리하지 못하고 또 어떤 날은 꼼꼼하지 못했습니다. 
하지만 완독이라는 목표를 스터디 덕분에 무사히 끝마칠 수 있을 것 같습니다! 😆

긴 시간동안 스터디 관리를 도맡아 해주신 스터디장님께 정말 감사의 말씀을 드립니다. 
스터디장님의 부지런함을 보면서 동기부여가 된 날도 참 많았습니다 😊

다음 시즌이 있다면 그 때는 책 내용뿐만 아니라 더 깊이 알아볼 수 있는 시간을 가지며 스터디에 참여하고 싶네요! 
다들 너무 수고하셨습니다 !! 👏🏻

**Comment by ccconac** [comment]
# 🏅 시즌 2 스터디 후기
* 이름 : 문소희
* 참여한 part : 1, 2, 3

### 1️⃣ 스터디 하면서 도움이 되었던 부분은?

독학을 할 때 체계적으로 범위를 끊어 두고 공부하는 것이 아니라 책에 맞춰 한 챕터씩 끊어 공부하다 보니, 어떤 챕터의 분량이 유독 많거나 적을 때가 있어 항상 공부를 하다 제풀에 지쳐 그만둔 적이 굉장히 많았습니다. 그래서 이 책을 완독해 보고 싶다는 생각만 하고 실행에 옮기지는 못하고 있었는데 마침 주마다 일정이 체계적으로 짜여 있는 스터디를 만나 무사히 완독을 할 수 있었습니다. 🎉 어느 하나에 치우치지 않고 그날의 공부에 열중할 수 있었다는 점에서 너무 만족스러웠어요. 

또한 가끔은 합리화를 하고 쉬어 가고 싶을 때도 있었는데 완주 배지와 같은 시스템으로 인해 더욱 열중할 수 있었던 것 같습니다. 그날 정리를 하지 못하고 잠들면 새벽 4시에라도 일어나서 그날 주어진 분량은 꼭 정리하고 다시 잠에 들 정도로요. 이런 좋은 스터디와 시스템을 고안해 주신 철원 님께 감사드립니다. 🙇🏻‍♀️

### 2️⃣ 스터디 하면서 힘들었던 부분은?

잘 짜여진 스터디 시스템 덕분에 스터디 관련으로 힘든 부분은 없었습니다. 대신 스터디를 fork해서 진행하다 보니 branch와 merge 관련해서 제 지식이 얕았던 것을 알게 되었고, 이에 관해 추가적으로 공부하는 시간을 가지게 되어 오히려 좋았어요. 👏 

### 3️⃣ 스터디하고나서 느낀점 및 후기

Deep Dive는 자바스크립트를 사용하고 있으면서도 언어에 대한 이해도가 굉장히 부족한 것 같아 서적을 찾아보고 있던 중에 발견한 서적입니다. 앞선 문항에서도 이야기했지만 가장 큰 문제는 960 페이지라는 압도적인 분량 때문에 시작하지 못했던 거였어요. 그렇게 나중에 시간이 되면 읽어 봐야겠다고 생각만 하고 미루다 Deep Dive 스터디를 발견하게 되어서 프론트엔드 공부의 가장 큰 목표였던 Deep Dive 1회독을 완료할 수 있었던 것 같습니다. 

1회독을 하면서 굉장히 얕은 지식으로 섣부르게 언어를 사용하고 있었던 것 같다고 느꼈습니다. 사용하지 않다 보니 지금까지 한 번도 들어보지 못한 개념도 있었고, 평소 빈번하게 사용하던 함수나 객체들도 어떠한 원리로 동작하고 있는지 이번에 처음 알게 되었어요. 이제는 이런 개념을 처음 들어보는 누군가에게 쉽게 설명해 줄 수 있을 때까지 여러 번 읽어 보며 깊게 이해해 볼 예정입니다. 👀 읽어 보고 나니 관련한 기술 블로그도 훨씬 잘 읽혀서 너무 좋아요.

### 4️⃣ 하고 싶은 말 & 다음에는 이런 게 추가/보완되었으면 좋겠어요

다루지 않는 부분에 대한 스케줄표 또한 제공해 주시고 있지만, 기간이 조금 더 길어지더라도 현재 일정에는 포함되지 않은 내용들도 스터디에서 다룰 수 있다면 좋을 것 같습니다. 저는 2회독을 진행하면서 포함되지 않은 내용까지 별도로 진행해 보려고 합니다. 물론 스터디에서 거의 모든 내용을 다룰 수 있었어서 너무 좋았어요.

그동안 좋은 스터디 운영해 주신 점에 대해 다시 한번 감사드립니다! 🙇🏻‍♀️
